backend/src/Utils/Functions.hs

```diff
@@ -9,8 +15,34 @@ whenJust :: Monad m => Maybe a -> (a -> m ()) -> m ()
 whenJust (Just x) f = f x
 whenJust Nothing _ = return ()
 
-
 -- | Given a @NominalDiffTime@, returns a multiple of the @NominalDiffTime@
 times :: NominalDiffTime -> Int -> NominalDiffTime
 times dt n = iterate (+ dt) dt !! n
 
+dateToDay :: Date -> Day
+dateToDay (Date d) = d
+
+userToEntity :: User -> UserEntity
+userToEntity u = UserEntity (name u) (username u) (password u)
+
+occupancyToEntity :: Occupancy -> UserEntityId -> OccupancyEntity
+occupancyToEntity o = OccupancyEntity (pack $ oTitle o) (oStart o) (oEnd o)
+
+entityToUser :: Entity UserEntity -> User
+entityToUser e = do
+   let id  = fromSqlKey $ entityKey e
+   let val = entityVal e
+   User 
+    (fromIntegral id) 
+    (userEntityName val) 
+    (userEntityUsername val) 
+    (userEntityPassword val)
+
+entityToOccupancy :: Entity OccupancyEntity -> Occupancy
+entityToOccupancy e =do 
+  let val = entityVal e
+  
+  Occupancy 
+    (show $ occupancyEntityTitle val) 
+    (occupancyEntityStart val) 
+    (occupancyEntityEnd val)
```

Comment:
These kinds of mappings might be moved to a typeclass, but that's not a priority